src: allow to negate boolean CLI flags

targos opened the pull request:
This change allows all boolean flags to be negated using the `--no-`
prefix.
Flags that are `true` by default (for example `--deprecation`) are
still documented as negations.
With this change, it becomes possible to easily flip the default
value of a boolean flag and to override the value of a flag passed
in the `NODE_OPTIONS` environment variable.

Co-authored-by: Anna Henningsen <anna@addaleax.net>

targos commented on src/node_options-inl.h:
```diff
@@ -23,12 +23,14 @@ template <typename Options>
-                                       OptionEnvvarSettings env_setting) {
+                                       OptionEnvvarSettings env_setting,
+                                       bool default_is_true) {
   options_.emplace(name,
                    OptionInfo{kBoolean,
                               std::make_shared<SimpleOptionField<bool>>(field),
                               env_setting,
-                              help_text});
+                              help_text,
+                              default_is_true});
```
I think I'm doing something wrong here, but was not able to find what.

Even when `AddOption` is called with `default_is_true = true`, the value of `OptionInfo.default_is_true` will always be false.

addaleax replied:
I think you’ll also want to add the field to the relevant `OptionsParser<Options>::Convert` overload here

targos commented:
/cc @addaleax

> to override the value of a flag passed in the NODE_OPTIONS environment variable.

I'll write a test for this when the PR is ready.

mhdawson reviewed (approved):
LGTM

targos commented:
I added three commits:

- `process.allowedNodeEnvironmentFlags` now includes the negated versions of the flags
- Restored the position of the flags documented as negated in `node --help`
- Added tests

targos commented:
@mhdawson @BridgeAR would you like to have another look?

mhdawson reviewed (approved):
Still LGTM

targos commented:
Landed in 1bbe66f43259
